packages/account/src/BiconomySmartAccount.ts

```diff
@@ -412,8 +420,22 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
         }
 
         // Requesting to update gas limits again (especially callGasLimit needs to be re-calculated)
-        finalUserOp = await this.estimateUserOpGas(finalUserOp)
-        Logger.log('userOp after estimation ', finalUserOp)
+        try {
+          finalUserOp = await this.estimateUserOpGas(finalUserOp)
+          const cgl = ethers.BigNumber.from(finalUserOp.callGasLimit)
+          if (finalUserOp.callGasLimit && cgl.lt(ethers.BigNumber.from('21000'))) {
```

Comment:
why we are adding this if statement ?
`if (finalUserOp.callGasLimit && cgl.lt(ethers.BigNumber.from('21000')))`

should not we just update finalUserOp callData with newCallData before calling estimateUserOpGas ?

```
finaluserOp.callData = newCallData
finalUserOp = await this.estimateUserOpGas(finalUserOp)
```

Reply:
its already updated above then sent for estimation. If estimation still returns less cgl then send earlier op with updated calldata.